### DISCUSSION TITLE
Clipboard usage in Rich ?

### DISCUSSION THREAD

**Discussion by gkzsolt** [Textualize/rich#2283] [Q&A] [answered]
Hi,
I am looking for a few days to Rich and Textual, and they look impressive. However there is something what bothers me and I'm afraid it will hinder their acceptance in my projects: clipboard. I've noticed that rich rendered text cannot be copied to, or pasted from the clipboard, and interestingly I did not find any reference to this. However, it is very much used in daily terminal sessions (at least in Linux), it's something I now realize. Is there a way to include clipboard usage into Rich/Textual, or there is a fundamental impediment?
(I don't know if this belongs here or to the textual discussion list, pls. correct me if I am mistaken.)
Note: I am not so obsessed with formatted text on the clipboard, almost all questions seems to be related to this. On Linux we use normal, unformatted text to/from the clipboard every day. I would be very happy to have only this.

**Comment by USLTD** [comment, marked as answer]
What do you mean it's can't be copied? Do you mean that things like colors and styles can't be copied? If yes then it's not rich's fault since it depends on Terminal Emulator used to allow for styles to be copied too. Normal behavior is to only copy characters.

And willmcgugan or anyone with post removal ability. If you are going to delete this answer too at least give me a reason why 😑

**Reply by willmcgugan** [reply]
@UltraStudioLTD Your replies are sometimes unhelpful. Suggest you try to clear up misunderstandings or offer suggestions, rather than simply telling people they are wrong.

**Reply by USLTD** [reply]
@willmcgugan Alright.

**Reply by gkzsolt** [reply]
I was referring to Rich and Textual, but tried only Textual.

**Comment by willmcgugan** [comment]
@gkzsolt The clipboard is handled by the terminal emulator, and Rich doesn't do anything to prevent that. You should be able to copy text just fine.

Textual runs in "alternate mode" (like all other full-screen terminal apps) which prevents copying by default, but you can hold down ctrl to select text. Textual doesn't have text input at the moment so there is nothing to paste in to, but when the new version is released paste will work just fine.

**Reply by gkzsolt** [reply]
Thanks, @willmcgugan (I am quite sad about the news). How can I hold down ctrl to select text? I tried it - dragging, or double-clicking while holding down Ctrl, but it does not work. It does not highlight the text either. I am using terminal emulators in Linux.

**Reply by gkzsolt** [reply]
Btw, it occured to me that if you had mouse dragging, or at least double-click event handling in Textual, this could be accomplished (implemented). But, to my surprise, it seems to me these events are not implemented. Why?
